sdk/storage/Azure.Storage.DataMovement.Blobs/tests/PageBlobDirectoryStartTransferDownloadTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+extern alias DMBlobs;
+extern alias BaseBlobs;
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using BaseBlobs::Azure.Storage.Blobs;
+using BaseBlobs::Azure.Storage.Blobs.Models;
+using BaseBlobs::Azure.Storage.Blobs.Specialized;
+
+namespace Azure.Storage.DataMovement.Blobs.Tests
+{
+    [DataMovementBlobsClientTestFixture]
+    public class PageBlobDirectoryStartTransferDownloadTests
+        : BlobDirectoryStartTransferDownloadTestBase
+    {
+        public PageBlobDirectoryStartTransferDownloadTests(
+            bool async,
+            BlobClientOptions.ServiceVersion serviceVersion)
+        : base(async, serviceVersion)
+        {
+        }
+
+        protected override async Task CreateBlobClientAsync(
+            BlobContainerClient container, string blobName, Stream contents, CancellationToken cancellationToken = default)
+        {
+            PageBlobClient pageBlobClient = container.GetPageBlobClient(blobName);
+            await pageBlobClient.CreateAsync(512 * 10);
+            if (contents != default)
+            {
+                if (contents.Length == 0 || contents.Length % 512 != 0)
```

Comment:
we should just throw an exception if someone puts in a content.Length is not divisible by 512

Reply:
Given tests are likely shared across blob types, some of which may want to test with odd sizes, I'd say we may want to `Assert.Inconclusive()` with a message about page blob 512 multiples.

Reply:
I've already set the precedent to assert an error when a length isn't in 512 increments.
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.DataMovement.Blobs/tests/PageBlobStartTransferCopyTests.cs#L107

And seems to be not causing an issue.

Yes from a high level it's causing the same issue where an exception is being thrown. However this exception we're throwing is more clear and will be resolved by whoever is writing test code. This is more for new people onboarding (or brain fart on the fact page blobs need to be 512).

Yes just change the content length to be 512 instead of 100. I'd rather the test be clear on what the length should be, and not mask the fact that the length is changing.

Reply:
If we can successfully test other blob sizes with other blob types then I have no issue with this. It just seems like there are shared test implementations and we may be locking ourselves out of some testing scenarios if we fail tests on this. There are other ways around that if we end up encountering it later on, so I'm not too worried.

Reply:
I like Jocelyn's suggestion of Assert.Inconclusive() (was not aware we can do this) so all other blobs and file shares can still use the old sizes

Reply:
I'll revert this [change ](https://github.com/Azure/azure-sdk-for-net/pull/46826/files/598773e9d5a1cf7e33f455de11d0fc3d280da075#diff-1b866cd970a71fd7bd86608d263920f018240125dfa0f287c48de29cb2609e68) too and use Assert.Inconclusive()